.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,164 @@
+        default: false
+        required: false
+        type: boolean
+
+jobs:
+  # Define the matrix of deltakit subprojects that have been modified in the PR and
+  # should be tested. This is done by looking at tags, so label_pr.yml should have done
+  # its job before.
+  # Because there is no easy way to ensure that label_pr.yml runs before this workflow,
+  # this job returns all the packages when no tags are set.
+  define-project-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      projects: ${{ steps.json_labels.outputs.projects }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v6
+      - name: Set up Python
+        uses: actions/setup-python@v6
+        with:
+          python-version: "3.10"
+      - name: Outputs modified packages that needs to be tested.
+        id: json_labels
+        run: python tools/transform_labels.py
+        env:
+          projects: ${{ join(github.event.pull_request.labels.*.name, ',') }}
+
+  # Build deltakit wheels.
+  build:
+    runs-on: ubuntu-latest
+    needs: define-project-matrix
+    strategy:
+      matrix:
+        projects: ${{ fromJSON(needs.define-project-matrix.outputs.projects) }}
+    steps:
+      - uses: actions/checkout@v6
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v7
+
+      - name: Build deltakit
+        run: uv build --no-sources --package ${{ matrix.projects.project }}
+
+      - name: Upload ${{ matrix.projects.project }} wheels
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.projects.project }}-wheel
+          path: ./dist/*.whl
+          retention-days: 7
+
+  merge_upload_artefacts:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Merge Artefacts
+        uses: actions/upload-artifact/merge@v4
+        with:
+          name: all-deltakit-wheels
+          pattern: deltakit*-wheel
+
+  # Test the built wheels.
+  tests:
+    needs: [merge_upload_artefacts, define-project-matrix]
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12', '3.13']
+        # python-version: ['3.12', '3.13']
+        dependency-resolution: ['highest', 'lowest-direct']
+        projects: ${{ fromJSON(needs.define-project-matrix.outputs.projects) }}
+    name: tests (${{ matrix.projects.project }}, ${{ matrix.python-version }}, ${{matrix.dependency-resolution}})
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v6
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v7
+
+      - name: Download wheels
+        uses: actions/download-artifact@v4
+        with:
+          name: all-deltakit-wheels
+          path: ./dist
+
+      - name: Setup venv
+        run: |
+          uv sync --package ${{ matrix.projects.project }} \
+            --python ${{ matrix.python-version }} \
+            --resolution ${{matrix.dependency-resolution}} \
+            --group=test \
+            --find-links ./dist/
```

Comment:
Even though the line will be terrible, I think it would be better to have a one liner here (without the `\`). The reason behind that is that, when one test job fails, it is convenient to be able to copy-paste this `uv sync` line to replicate the environment locally, in order to replicate the error locally too. But having that on multiple lines makes the copy-paste process harder than it needs to be.